roms_tools/plot.py

```diff
@@ -597,13 +631,464 @@ def _add_field_to_ax(
+
+    if "straddle" not in grid_ds.attrs:
+        raise AttributeError("Grid dataset must have a 'straddle' attribute.")
+    straddle = grid_ds.attrs["straddle"]
+
+    # Get horizontal dimensions and grid location
+    horizontal_dims_dict = {
+        "rho": {"eta": "eta_rho", "xi": "xi_rho"},
+        "u": {"eta": "eta_rho", "xi": "xi_u"},
+        "v": {"eta": "eta_v", "xi": "xi_rho"},
+    }
+    for loc, horizontal_dims in horizontal_dims_dict.items():
+        if all(dim in field.dims for dim in horizontal_dims.values()):
+            break
+
+    # Convert relative to absolute indices
+    def _get_absolute_index(idx, field, dim_name):
+        index = field[dim_name].isel(**{dim_name: idx}).item()
+        return index
+
+    if eta is not None and eta < 0:
+        eta = _get_absolute_index(eta, field, horizontal_dims["eta"])
+    if xi is not None and xi < 0:
+        xi = _get_absolute_index(xi, field, horizontal_dims["xi"])
+    if s is not None and s < 0:
+        s = _get_absolute_index(s, field, "s_rho")
+
+    # Set spatial coordinates
+    if use_coarse_grid:
+        if loc != "rho":
+            raise ValueError("If using coarse grid, can only plot on rho-points.")
+        lat_deg = grid_ds["lat_coarse"].rename(
+            {"eta_coarse": "eta_rho", "xi_coarse": "xi_rho"}
+        )
+        lon_deg = grid_ds["lon_coarse"].rename(
+            {"eta_coarse": "eta_rho", "xi_coarse": "xi_rho"}
+        )
+        mask = grid_ds["mask_coarse"].rename(
+            {"eta_coarse": "eta_rho", "xi_coarse": "xi_rho"}
+        )
+    else:
+        lat_deg = grid_ds[f"lat_{loc}"]
+        lon_deg = grid_ds[f"lon_{loc}"]
+        mask = grid_ds[f"mask_{loc}"]
+
+    if straddle:
+        lon_deg = xr.where(lon_deg > 180, lon_deg - 360, lon_deg)
+    if lon is not None:
+        lon = normalize_longitude(lon, straddle)
+
+    field = field.assign_coords({"lon": lon_deg, "lat": lat_deg})
+
+    # Mask the field
+    field = field.where(mask)
+
+    # Assign eta and xi as coordinates
+    coords_to_assign = {dim: field[dim] for dim in horizontal_dims.values()}
+    field = field.assign_coords(**coords_to_assign)
+
+    # Remove horizontal boundary if desired
+    slice_dict = {
+        "rho": {"eta_rho": slice(1, -1), "xi_rho": slice(1, -1)},
+        "u": {"eta_rho": slice(1, -1), "xi_u": slice(1, -1)},
+        "v": {"eta_v": slice(1, -1), "xi_rho": slice(1, -1)},
+    }
+    if not include_boundary:
+        field = field.isel(**slice_dict[loc])
+
+    # Prepare figure title
+    if "abs_time" in field.coords:
+        formatted_time = np.datetime_as_string(field.abs_time.values, unit="m")
+        title = f"time: {formatted_time}"
+    elif "ntides" in field.coords:
+        title = f"constituent: {field.ntides.values.item().decode('utf-8')}"
+    elif "long_name" in field.attrs:
+        title = field.long_name
+    else:
+        title = ""
+
+    # Slice the field horizontally as desired
```

Comment:
ditto on comment preceding function. move to a docstring or consider removing/renaming function